### ISSUE TITLE
Fix all lints currently disabled in scripts/sync.sh

### ISSUE THREAD

**Issue by ia0** [google/wasefire#565]
We want to enable the following lints:
- [ ] `rust.elided-lifetimes-in-paths`
- [ ] `rust.missing-debug-implementations` (not sure if good thing)
- [ ] `rust.missing-docs` for all published libraries
- [ ] `rust.unreachable-pub`
- [ ] `rust.unused-results`

**Comment by devesh-2002** [comment]
May I take this issue?

**Comment by ia0** [comment]
Of course, that would be great. That's a good first issue. You can start by changing those 2 lines:

https://github.com/google/wasefire/blob/5da7a5e0de8545825ce6ca023c389f2e35926c02/scripts/sync.sh#L33-L34

into (I'm removing `missing-debug-implementations` because I'm not yet convinced it's a good thing to have):

```rust
  case $crate in
    prelude) add_lint $file warn rust.elided-lifetimes-in-paths ;;
  esac
```

Then run `./scripts/sync.sh` (note that you might need to run `./scripts/setup.sh` the very first time you clone the repository). This will add the lint to the prelude Cargo.toml. Then you can run `./test.sh` from the `crates/prelude` directory and fix any warning.

Once there are no warnings, you can create your first PR.

Then, you can add more and more crates to the case pattern, like:

```rust
  case $crate in
    board|prelude) add_lint $file warn rust.elided-lifetimes-in-paths ;;
  esac
```

And create one PR for each non-trivial crate (you can group multiple crates together if there's nothing or almost nothing to change).

The final state (i.e. when adding the last remaning crate) should simply be:

```rust
  add_lint $file warn rust.elided-lifetimes-in-paths
```

Thanks! Feel free to ask questions if you're blocked.

**Comment by devesh-2002** [comment]
Thank you @ia0, for the detailed instructions!

**Comment by devesh-2002** [comment]
@ia0 , Hello. I ran `./scripts/setup.sh` and then `./scripts/sync.sh`. But it gives me a duplicate key error. Would you please guide me in resolving this error?
```
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 3.36s
     Running `target/debug/xtask update-apis`
"cargo" "run" "--manifest-path=crates/xtask/crates/update-api/Cargo.toml" "--features=wasefire-applet-api-desc/full-api" "--" "--lang=assemblyscript" "--output=examples/assemblyscript/api.ts"
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.69s
     Running `target/debug/update-api --lang=assemblyscript --output=examples/assemblyscript/api.ts`
error: invalid table header
duplicate key `lints` in document root
  --> Cargo.toml:26:1
   |
26 | [lints]
   | ^
   |```

**Comment by ia0** [comment]
To debug:
- Could you push your working directory to a branch on your fork?
- What is your operating system?
- What is your version of `sed`? (`sed --version`)
- What is your version of `sh`? (`sh --version`)
- What is the path where wasefire is cloned?

I suspect that one of those 2 lines doesn't behave as expected in your case:

https://github.com/google/wasefire/blob/d32a3d98046b15e2295a71334462b1e5b12cd923/scripts/sync.sh#L30-L31
